src/cmd/build.rs

```diff
@@ -1105,6 +1218,40 @@ mod tests_ci_only {
         })
     }
 
+    #[test]
+    fn building_subcontract_must_work() {
+        with_new_subcontract_projects(
+            |manifest_path| {
+                let cmd = BuildCommand {
+                    build_all: true,
+                    package: None,
+                    manifest_path: Some(manifest_path),
+                    ..Default::default()
+                };
+                cmd.exec().expect("build failed");
+                Ok(())
+            },
+            1,
+        )
+    }
+
+    #[test]
+    fn building_many_subcontracts_must_work() {
```

Comment:
We also need a test for the `--all` flag

Reply:
because of https://github.com/paritytech/cargo-contract/commit/40548c3895fa6c537ed1e1a5f4b81c4835401aa9 I had to refactor the tests so they could fit into the [`build_tests!` macro](https://github.com/paritytech/cargo-contract/blob/29cc87a246a7a56e3df619802e82e376d5fee017/crates/cargo-contract/src/cmd/build/tests.rs#L44).

so [I dropped `with_new_subcontract_projects`](https://github.com/paritytech/cargo-contract/pull/647/commits/7315f4c7fa9b5a7c2fde4c5c1d53ae8203700c72) from `util::tests`.

anyways, I don't understand the suggestion here.
the `BuildCommand` sets `build_all: true` (both pre and post refactor), so that is essentially testing the `--all` flag.

am I missing something?

Reply:
ohh I think I get what you mean:

we should test both the `--all` and `--package` flags.

although the code did cover the `--all` flag like mentioned in the previous comment, there wasn't a test for the `--package`

that should be fixed now, as we have:
- `building_package_must_work`
- `building_all_must_work`